x/session/keeper/session_hydrator_test.go

```diff
@@ -86,20 +91,33 @@ func TestSession_HydrateSession_Metadata(t *testing.T) {
 			name:        "blockHeight != sessionHeight",
 			blockHeight: 5,
 
+			errExpected:                 false,
 			expectedNumBlocksPerSession: 4,
 			expectedSessionNumber:       1,
 			expectedSessionStartBlock:   4,
 		},
+		{
+			name:        "blockHeight > contextHeight",
+			blockHeight: 9001, // block height over 9000 is too height given that the context height is 100
+
+			errExpected: true,
+		},
 	}
 
 	appAddr := keepertest.TestApp1Address
 	serviceId := keepertest.TestServiceId1
 	sessionKeeper, ctx := keepertest.SessionKeeper(t)
+	ctx = ctx.WithBlockHeight(100) // provide a sufficiently large block height to avoid errors
 
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
 			sessionHydrator := keeper.NewSessionHydrator(appAddr, serviceId, tt.blockHeight)
 			session, err := sessionKeeper.HydrateSession(ctx, sessionHydrator)
+
+			if tt.errExpected {
+				require.Error(t, err)
```

Comment:
Wdyt  about making `tt.errExpected` an `error` type and doing something like:
```suggestion
				require.ErrorIs(t, tt.errExpected, err)
```